http-common/src/connector.rs

```diff
@@ -31,6 +31,52 @@ pub enum Incoming {
     Unix(tokio::net::UnixListener),
 }
 
+impl Incoming {
+    pub async fn serve<H>(&mut self, server: H) -> std::io::Result<()>
+    where
+        H: hyper::service::Service<
+                hyper::Request<hyper::Body>,
+                Response = hyper::Response<hyper::Body>,
+                Error = std::convert::Infallible,
+            > + Clone
+            + Send
+            + 'static,
+        <H as hyper::service::Service<hyper::Request<hyper::Body>>>::Future: Send,
+    {
+        match self {
+            Incoming::Tcp(listener) => loop {
+                let (tcp_stream, _) = listener.accept().await?;
+
+                // TCP is available in test builds only (not production). Assume current user is root.
+                let server = crate::uid::UidService::new(0, server.clone());
+                tokio::task::spawn(async move {
+                    if let Err(http_err) = hyper::server::conn::Http::new()
+                        .serve_connection(tcp_stream, server)
+                        .await
+                    {
+                        eprintln!("Error while serving HTTP connection: {}", http_err);
```

Comment:
`s/eprintln/log::info/`

Reply:
`log::error` right?

Reply:
It's not an error from the server's POV. It'll happen if say the client closes their connection in the middle of a request. After all, isn't that why you're logging and swallowing it instead of bubbling it up?

Reply:
Yes we shouldn't break the accept loop. Logging here is sufficient.